src/aiida/cmdline/commands/cmd_init.py

```diff
@@ -0,0 +1,103 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""``verdi init`` command."""
+import pathlib
+
+import click
+
+from aiida.cmdline.commands.cmd_verdi import verdi
+from aiida.cmdline.utils import echo
+
+
+@verdi.command('init')
+@click.argument(
+    'directory', type=click.Path(exists=False, path_type=pathlib.Path, resolve_path=True), default=pathlib.Path.cwd()
+)
+@click.option(
+    '--from-archive',
+    type=click.Path(dir_okay=False, exists=True, resolve_path=True, path_type=pathlib.Path),
+    help='Mount the archive as a read-only profile instead of creating an empty profile.',
+)
+def verdi_init(directory, from_archive):
+    """Initialize a new AiiDA instance.
+
+    The instance is initialized in DIRECTORY, which defaults to the currrent working directory. A profile is
+    automatically created using the `core.sqlite_dos` storage backend and the localhost is configured as a computer.
+    """
+    import tempfile
+    import warnings
+
+    from aiida.common import exceptions
+    from aiida.manage import get_manager
+    from aiida.manage.configuration import create_profile, get_config, load_profile, reset_config, settings
+    from aiida.orm import Computer
+
+    # Unload the profile that was loaded by default and reset the configuration.
+    get_manager().unload_profile()
+    reset_config()
+
+    dirpath_config = directory / settings.DEFAULT_CONFIG_DIR_NAME
+
+    if dirpath_config.is_dir():
+        echo.echo_critical(f'A configuration directory already exists in `{directory}`.')
+
+    with warnings.catch_warnings():
+        warnings.simplefilter('ignore')
+        settings.set_configuration_directory(dirpath_config)
+
+    echo.echo_success(f'Initialized new AiiDA instance in `{dirpath_config}`.')
+
+    # Now load the config causing the configuration file to be created.
+    config = get_config(create=True)
+    dirpath_loaded = pathlib.Path(config.filepath).parent
+    assert dirpath_loaded == dirpath_config, f'Filepath of loaded config `{dirpath_loaded}` != `{dirpath_config}`.'
+
+    if from_archive:
+        storage_backend = 'core.sqlite_zip'
+        storage_config = {
+            'filepath': str(from_archive),
+        }
+    else:
+        storage_backend = 'core.sqlite_dos'
+        storage_config = {}
+
+    try:
+        profile = create_profile(
+            config,
+            name='init',
+            email='aiida@localhost',
```

Comment:
I would suggest exposing more args in the cmd, such as `profile_name`,`user name`,`email` etc.

Reply:
I can see the point for the user settings, making them optional flags. Is there really a need for the profile name though? Not that it would be a huge problem to add, just wondering why a user would want to change it. The whole point of this command is to _not_ have to think and configure but it just does the right thing. What profile name is chosen seems irrelevant, especially if this setup essentially encourages to use single-profile instances.

Reply:
Just a quick idea perhaps for a follow-up PR: it would be great to have something analogous to `~/.gitconfig` where these common configs could be stored permanently for a given user.
(But maybe I am getting carried away here :-)

Reply:
Thanks for the explanation. I agree that the profile name is not important here.